R/helpers.R

```diff
@@ -204,3 +204,22 @@ format_dilutions <- function(dilutions, dilution_values, sample_types) {
 dilution_to_rau <- function(predicted_dilution) {
   return(predicted_dilution * 1e6)
 }
+
+#' @title Check if the vector is monotically decreasing
+#'
+#' @param x (`numeric()`) Vector of numeric values
+#'
+#' @return (`logical(1)`) `TRUE` if the vector is monotonically decreasing, `FALSE` otherwise
+#'
+is.decreasing <- function(x) {
+  stopifnot(is.numeric(x) || is.null(x))
+  if (any(is.na(x))) {
```

Comment:
if this function is gonna be used only in `handle_high_dose_hook` then as a user I would prefer to see different error message. I know that this check fits here but I would move it to `handle_high_dose_hook` and there gave it different, more specific error message such as "NA values detected in the dilutions while handling high dose hook" or something similar

Reply:
I guess such a situation should not happen. In general it would mean that a standard curve sample has a missing dilution value

Reply:
But I'd love to see a test in which we process a plate with high-hook

Reply:
I suggest to approve these changes. In general, if there were a problem with NA dilutions, it would also affect different parts of model creation. There are no specific exceptions thrown that could help track if there is standard curve sample without dilution.

I think, if @nizwant agrees we can approve these changes and optionally add this problem as a new issue, but I don't think that this is necessary.

Reply:
I tried to manipulate the plate object so that the standard curve sample would not have the dilution value, and it turns out that then it is read as a test sample instead. I am not sure how to enforce the behaviour that @nizwant wants to avoid other than through manual modification of plate object data.

Reply:
then it's fine